apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/DraggableLinkButton.tsx

```diff
@@ -0,0 +1,250 @@
+import type {
+  DraggableProvidedDraggableProps,
+  DraggableProvidedDragHandleProps,
+} from "@hello-pangea/dnd"
+import type { JsonSchema, UISchemaElement } from "@jsonforms/core"
+import {
+  Box,
+  Flex,
+  forwardRef,
+  HStack,
+  Icon,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Skeleton,
+  Text,
+  VStack,
+} from "@chakra-ui/react"
+import { computeChildLabel } from "@jsonforms/core"
+import { useJsonForms } from "@jsonforms/react"
+import { IconButton, Menu } from "@opengovsg/design-system-react"
+import { getResourceIdFromReferenceLink } from "@opengovsg/isomer-components"
+import {
+  BiDotsHorizontalRounded,
+  BiGridVertical,
+  BiPencil,
+  BiSolidErrorCircle,
+  BiTrash,
+} from "react-icons/bi"
+
+import type { LinkTypesWithHrefFormat } from "../../../LinkEditor/constants"
+import Suspense from "~/components/Suspense"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { sitePageSchema } from "~/pages/sites/[siteId]"
+import { trpc } from "~/utils/trpc"
+import { LINK_TYPES } from "../../../LinkEditor/constants"
+import { getLinkHrefType } from "../../../LinkEditor/utils"
+import { parseHref } from "./utils/parseHref"
+
+interface SuspendableLabelProps {
+  siteId: number
+  resourceId: string
+}
+const SuspendableLabel = ({ siteId, resourceId }: SuspendableLabelProps) => {
+  const [{ fullPermalink }] =
+    trpc.resource.getWithFullPermalink.useSuspenseQuery({
+      siteId,
+      resourceId,
+    })
+
+  return (
+    <Text
+      noOfLines={1}
+      textStyle="caption-2"
+      textColor="interaction.support.placeholder"
+    >{`/${fullPermalink}`}</Text>
+  )
+}
+
+interface DraggableLinkButtonProps {
+  draggableProps: DraggableProvidedDraggableProps
+  dragHandleProps: DraggableProvidedDragHandleProps | null
+  setSelectedIndex: (selectedIndex?: number) => void
+  onDeleteItem: () => void
+  isError: boolean
+  index: number
+  path: string
+  schema: JsonSchema
+  uischema: UISchemaElement
+}
+
+const DraggableLinkButton = forwardRef<DraggableLinkButtonProps, "div">(
+  (
+    {
+      draggableProps,
+      dragHandleProps,
+      setSelectedIndex,
+      onDeleteItem,
+      isError,
+      index,
+      path,
+      schema,
+      uischema,
+    },
+    ref,
+  ) => {
+    const ctx = useJsonForms()
+    const label = computeChildLabel(
+      ctx.core?.data,
+      path,
+      "",
+      schema,
+      ctx.core?.schema ?? {},
+      ctx.i18n?.translate ?? ((s) => s),
+      uischema,
+    )
+    const url = computeChildLabel(
+      ctx.core?.data,
+      path,
+      "url",
+      schema,
+      ctx.core?.schema ?? {},
+      ctx.i18n?.translate ?? ((s) => s),
+      uischema,
+    )
+
+    const { siteId } = useQueryParse(sitePageSchema)
+    const linkType = getLinkHrefType(url)
+    const displayedHref = parseHref(url, linkType as LinkTypesWithHrefFormat)
+
+    return (
+      <Box my="0.25rem" ref={ref} {...draggableProps} w="full">
+        <Box
```

Comment:
comment: not important now, but i thought it might be good to go through a hook like `useDraggable` to keep the styles in sync.

as an example,

```ts
const { getDraggableStyling } = useDraggable()

return <Box {...getDraggableStyling()} />
```